internal/cmd/gocdk: serve crashes immediately

@vangent Yup, I'm still filing issues. 🙂

### Describe the bug

Running `gocdk serve` crashes during `biome apply` due to a nil pointer panic.

### To Reproduce

`gocdk serve`

### Expected behavior

A server should be running.

### Actual behavior

```
gocdk: Preparing to serve dev...
gocdk: Checking to see if resource changes are needed...
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x2 pc=0xcbcaf0]

goroutine 12 [running]:
main.doBiomeApplyOrDestroy(0x11bb640, 0xc000035880, 0xc0002df260, 0xfd049c, 0x3, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
        /home/light/pkg/mod/gocloud.dev/internal/cmd/gocdk@v0.0.0-20190724210842-d2cfe8b44225/biome.go:329 +0x240
main.biomeApply(0x11bb640, 0xc000035880, 0xc0002df260, 0xfd049c, 0x3, 0x0, 0x0, 0x0)
        /home/light/pkg/mod/gocloud.dev/internal/cmd/gocdk@v0.0.0-20190724210842-d2cfe8b44225/biome.go:273 +0x86
main.serveBuildLoop(0x11bb640, 0xc000035880, 0xc0002df260, 0xc0000cb6d0, 0xc00031a930, 0xc0000355c0, 0x0, 0x0)
        /home/light/pkg/mod/gocloud.dev/internal/cmd/gocdk@v0.0.0-20190724210842-d2cfe8b44225/serve.go:144 +0x2f0
main.serve.func2(0x8, 0x100f660)
        /home/light/pkg/mod/gocloud.dev/internal/cmd/gocdk@v0.0.0-20190724210842-d2cfe8b44225/serve.go:96 +0x57
golang.org/x/sync/errgroup.(*Group).Go.func1(0xc00031a990, 0xc00031a9f0)
        /home/light/pkg/mod/golang.org/x/sync@v0.0.0-20190423024810-112230192c58/errgroup/errgroup.go:57 +0x57
created by golang.org/x/sync/errgroup.(*Group).Go
        /home/light/pkg/mod/golang.org/x/sync@v0.0.0-20190423024810-112230192c58/errgroup/errgroup.go:54 +0x66
```

### Version

gocdk CLI built at d2cfe8b442254eaedc43ce1060b65846b91f3598

### Additional context

The issue appears to be that `gocdk serve` calls `biomeApply` with `nil` options:

https://github.com/google/go-cloud/blob/d2cfe8b442254eaedc43ce1060b65846b91f3598/internal/cmd/gocdk/serve.go#L144

and that when building the Terraform arguments, `doBiomeApplyOrDestroy` does not check for `nil` before dereferencing the options:

https://github.com/google/go-cloud/blob/d2cfe8b442254eaedc43ce1060b65846b91f3598/internal/cmd/gocdk/biome.go#L329

--- a/internal/cmd/gocdk/biome.go
+++ b/internal/cmd/gocdk/biome.go
@@ -72,6 +72,7 @@ information needed to deploy.`,
 	}
 	biomeCmd.AddCommand(listCmd)
 
+	var input bool
 	var applyOpts biomeApplyOptions
 	applyCmd := &cobra.Command{
 		Use:   "apply <biome name>",
@@ -82,10 +83,11 @@ configuration (e.g., creating or updating cloud resources).
 Runs "terraform init" followed by "terraform apply".`,
 		Args: cobra.ExactArgs(1),
 		RunE: func(_ *cobra.Command, args []string) error {
-			return biomeApply(ctx, pctx, args[0], &applyOpts)
+			applyOpts.DisableInput = !input
+			return biomeApply(ctx, pctx, args[0], applyOpts)
 		},
 	}
-	applyCmd.Flags().BoolVar(&applyOpts.Input, "input", true, "ask for input for Terraform variables if not directly set")
+	applyCmd.Flags().BoolVar(&input, "input", true, "ask for input for Terraform variables if not directly set")
 	applyCmd.Flags().BoolVar(&applyOpts.Verbose, "verbose", false, "true to print all output from Terraform")
 	applyCmd.Flags().BoolVar(&applyOpts.AutoApprove, "auto-approve", false, "true to auto-approve resource changes")
 	biomeCmd.AddCommand(applyCmd)
@@ -100,7 +102,7 @@ Runs "terraform init" followed by "terraform apply".`,
 Runs "terraform destroy".`,
 		Args: cobra.ExactArgs(1),
 		RunE: func(_ *cobra.Command, args []string) error {
-			return biomeCleanup(ctx, pctx, args[0], &cleanupOpts, deleteBiome)
+			return biomeCleanup(ctx, pctx, args[0], cleanupOpts, deleteBiome)
 		},
 	}
 	cleanupCmd.Flags().BoolVar(&deleteBiome, "delete", false, "delete the entire biome")
@@ -259,24 +261,25 @@ func biomeList(ctx context.Context, pctx *processContext) error {
 }
 
 // biomeApplyOptions holds options for "biome apply" or "biome cleanup".
+// The zero value is the default set of options.
 type biomeApplyOptions struct {
-	Verbose     bool // show all Terraform output
-	AutoApprove bool // auto-approve Terraform changes
-	Input       bool // allow Terraform prompts for input
+	Verbose      bool // show all Terraform output
+	AutoApprove  bool // auto-approve Terraform changes
+	DisableInput bool // disable Terraform prompts for input
 }
 
 // biomeApply implements the "biome apply" subcommand.
 //
 // It runs "terraform init", "terraform plan", and "terraform apply" for the
 // named biome.
-func biomeApply(ctx context.Context, pctx *processContext, biome string, opts *biomeApplyOptions) error {
+func biomeApply(ctx context.Context, pctx *processContext, biome string, opts biomeApplyOptions) error {
 	if _, err := doBiomeApplyOrDestroy(ctx, pctx, biome, opts, false); err != nil {
 		return xerrors.Errorf("biome apply %s: %w", biome, err)
 	}
 	return nil
 }
 
-func doBiomeApplyOrDestroy(ctx context.Context, pctx *processContext, biome string, opts *biomeApplyOptions, destroy bool) (string, error) {
+func doBiomeApplyOrDestroy(ctx context.Context, pctx *processContext, biome string, opts biomeApplyOptions, destroy bool) (string, error) {
 	moduleRoot, err := pctx.ModuleRoot(ctx)
 	if err != nil {
 		return "", err
@@ -326,7 +329,7 @@ func doBiomeApplyOrDestroy(ctx context.Context, pctx *processContext, biome stri
 
 	// First, run "terraform init". It's needed the first time, and anytime
 	// a new Terraform provider is added, so it's safest to just always run it.
-	inputArg := fmt.Sprintf("-input=%v", opts.Input)
+	inputArg := fmt.Sprintf("-input=%t", !opts.DisableInput)
 	c := pctx.NewCommand(ctx, biomePath, "terraform", "init", inputArg)
 	c.Env = overrideEnv(c.Env, "TF_IN_AUTOMATION=1")
 	if out, err := runTerraform(c); err != nil {
@@ -426,7 +429,7 @@ func doBiomeApplyOrDestroy(ctx context.Context, pctx *processContext, biome stri
 // for the named biome.
 //
 // If deleteBiome is true, it also deletes the biome subdirectory.
-func biomeCleanup(ctx context.Context, pctx *processContext, biome string, opts *biomeApplyOptions, deleteBiome bool) error {
+func biomeCleanup(ctx context.Context, pctx *processContext, biome string, opts biomeApplyOptions, deleteBiome bool) error {
 	biomePath, err := doBiomeApplyOrDestroy(ctx, pctx, biome, opts, true)
 	if err != nil {
 		return xerrors.Errorf("biome cleanup %s: %w", biome, err)

--- a/internal/cmd/gocdk/deploy.go
+++ b/internal/cmd/gocdk/deploy.go
@@ -72,7 +72,7 @@ func deploy(ctx context.Context, pctx *processContext, biome, dockerImage string
 
 	// Run "biome apply".
 	if apply {
-		if err := biomeApply(ctx, pctx, biome, nil); err != nil {
+		if err := biomeApply(ctx, pctx, biome, biomeApplyOptions{}); err != nil {
 			return err
 		}
 	}

--- a/internal/cmd/gocdk/serve.go
+++ b/internal/cmd/gocdk/serve.go
@@ -141,7 +141,7 @@ func serveBuildLoop(ctx context.Context, pctx *processContext, logger *log.Logge
 	}()
 
 	// Apply Terraform configuration in biome.
-	if err := biomeApply(ctx, pctx, opts.biome, nil); err != nil {
+	if err := biomeApply(ctx, pctx, opts.biome, biomeApplyOptions{}); err != nil {
 		return err
 	}
 

